subsys/nrf_security/src/drivers/cracen/cracenpsa/src/ed25519.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <string.h>
+#include <sxsymcrypt/sha2.h>
+#include <silexpk/ed25519.h>
+#include <cracen/ec_helpers.h>
+#include <sxsymcrypt/hash.h>
+#include <cracen/mem_helpers.h>
+
+int ed25519_sign(const uint8_t *ed25519, char *signature, const uint8_t *message)
+{
+	char workmem[160];
+	struct sxhash hashopctx;
+	char pointr_buffer[64];
+	char *pointr = pointr_buffer;
+	int r;
+
+	/*Hash the private key, the digest is stored in the first 64 bytes of workmem*/
```

Comment:
Technically parts of this code is a duplicate of the `create_public_key` function further down. But as I believe you need the intermediate steps calculated in working memory anyways it makes more sense to have the duplication than to call `create_public_key` and then redo the operations.